Source/Core/Common/Arm64Emitter.cpp

```diff
@@ -784,6 +571,18 @@ void ARM64XEmitter::EncodeLogicalImmInst(u32 op, ARM64Reg Rd, ARM64Reg Rn, u32 i
           (DecodeReg(Rn) << 5) | DecodeReg(Rd));
 }
 
+void ARM64XEmitter::EncodeLogicalImmInst(u32 op, ARM64Reg Rd, ARM64Reg Rn, LogicalImm imm)
+{
+  ASSERT_MSG(DYNAREC, imm.valid, "Invalid logical immediate");
```

Comment:
Could you also add an assert if the width of the imm matches the width of Rn?

Reply:
I can add an assert that you don't use a 64-bit imm with a 32-bit register, but all other combinations of register size and imm width are valid.